src/reorderable/ReorderableListComponent.coffee

```diff
@@ -23,21 +24,24 @@ class ReorderableListComponent extends React.Component
     @state = {
       initialOrder: order
       order: order
+      listId: if @props.listId then @props.listId else uuid.v4()
     }
 
   componentWillReceiveProps: (nextProps) ->
     order = _.map nextProps.items, (item) => @props.getItemId(item)
     if not _.isEqual(order, @state.initialOrder)
       @setState(initialOrder: order, order: order)
 
+    if nextProps.listId
+      @setState(listId: nextProps.listId)
```

Comment:
Close.... it should use the same logic as constructor. Otherwise no list id will mean use the old one which could cause problems. Just `@setState(listId: if nextProps.listId then nextProps.listId else uuid.v4())`